steps/http/session.go

```diff
@@ -72,6 +74,12 @@ func (s *Session) URL() (*url.URL, error) {
 	return u, nil
 }
 
+// ConfigureHost configures the HTTP request Host.
```

Comment:
Instead of creating a new step, I would take the "Host" request header (if available) and apply it in SendHTTPRequest.